python/lsst/ts/eas/eas_csc.py

```diff
@@ -84,290 +85,223 @@ def __init__(
+                try:
+                    aperture_shutter = await dome_remote.tel_apertureShutter.aget(
+                        timeout=STD_TIMEOUT
+                    )
+                except TimeoutError:
+                    continue
+                shutter_closed = (
+                    aperture_shutter.positionActual[0] < 0.1
+                    and aperture_shutter.positionActual[1] < 0.1
+                )
+                self.log.debug(f"{aperture_shutter.positionActual=} {shutter_closed=}")
+
+                if not shutter_closed and (
+                    utils.current_tai() - self.last_vec04_time
+                    > self.config.vec04_hold_time
+                ):
+                    # Check windspeed threshold
+                    wind_threshold = self.average_windspeed < self.config.wind_threshold
+                    self.log.debug(f"VEC-04 operation demanded: {wind_threshold}")
+                    if wind_threshold != cached_wind_threshold:
+                        cached_wind_threshold = wind_threshold
+                        if wind_threshold:
+                            self.log.info("Turning on VEC-04 fan!")
+                            await hvac_remote.cmd_enableDevice.set_start(
+                                device_id=DeviceId.lowerDamperFan03P04
+                            )
+                        else:
+                            self.log.info("Turning off VEC-04 fan!")
+                            await hvac_remote.cmd_disableDevice.set_start(
+                                device_id=DeviceId.lowerDamperFan03P04
+                            )
+
+                if shutter_closed != cached_shutter_closed:
+                    cached_shutter_closed = shutter_closed
+                    ahus = (
+                        DeviceId.lowerAHU01P05,
+                        DeviceId.lowerAHU02P05,
+                        DeviceId.lowerAHU03P05,
+                        DeviceId.lowerAHU04P05,
+                    )
+                    if shutter_closed:
+                        # Enable the four AHUs
+                        self.log.info("Enabling HVAC AHUs!")
+                        for device in ahus:
+                            await hvac_remote.cmd_enableDevice.set_start(
+                                device_id=device
+                            )
+
+                        # Disable the VEC-04 fan
+                        self.log.info("Turning off VEC-04 fan!")
+                        await hvac_remote.cmd_disableDevice.set_start(
+                            device_id=DeviceId.lowerDamperFan03P04
+                        )
+                    else:
+                        self.log.info("Disabling HVAC AHUs!")
+                        for device in ahus:
+                            await hvac_remote.cmd_disableDevice.set_start(
+                                device_id=device
+                            )
+
+                await asyncio.sleep(HVAC_SLEEP_TIME)
```

Comment:
Don't forget to close the remotes before exiting from this method.

Reply:
That assumes the CSC is used as an async context manager which is not the case when running in the deployment container. I checked the method and it calls `self.close` but that method is not overridden in the EAS CSC code so I still think the remotes need to be closed manually.

Reply:
Yes, of course. Sorry that I missed that!